core/src/main/scala/kafka/controller/KafkaController.scala

```diff
@@ -1680,45 +1680,61 @@ class KafkaController(val config: KafkaConfig,
   private def processPartitionModifications(topic: String): Unit = {
     def restorePartitionReplicaAssignment(
       topic: String,
-      newPartitionReplicaAssignment: Map[TopicPartition, ReplicaAssignment]
+      restorePartitionReplicaAssignment: Map[TopicPartition, ReplicaAssignment]
     ): Unit = {
       info("Restoring the partition replica assignment for topic %s".format(topic))
 
-      val existingPartitions = zkClient.getChildren(TopicPartitionsZNode.path(topic))
-      val existingPartitionReplicaAssignment = newPartitionReplicaAssignment
-        .filter(p => existingPartitions.contains(p._1.partition.toString))
-        .map { case (tp, _) =>
-          tp -> controllerContext.partitionFullReplicaAssignment(tp)
-      }.toMap
-
       zkClient.setTopicAssignment(topic,
         controllerContext.topicIds.get(topic),
-        existingPartitionReplicaAssignment,
+        restorePartitionReplicaAssignment,
         controllerContext.epochZkVersion)
     }
 
     if (!isActive) return
     val partitionReplicaAssignment = zkClient.getFullReplicaAssignmentForTopics(immutable.Set(topic))
-    val partitionsToBeAdded = partitionReplicaAssignment.filter { case (topicPartition, _) =>
+    val newPartitionsToBeAdded = partitionReplicaAssignment.filter { case (topicPartition, _) =>
```

Comment:
> newPartitionsToBeAdded seems verbose since new and added are describing the same thing. So, it could just be partitionsToBeAdded. Similarly, oldPartitionsToBeModified could just be partitionsToBeModified.

You said is correct, I have revised it